docs/content/preview/releases/techadvisories/ta-26666.md

```diff
@@ -0,0 +1,92 @@
+---
+title: TA-26666
+headerTitle: Memory leak in workloads with foreign keys or serializable reads
+headcontent: 02 September 2025
+type: docs
+showRightNav: true
+cascade:
+  unversioned: true
+menu:
+  preview_releases:
+    identifier: ta-26666
+    weight: 1
+rightNav:
+  hideH2: true
+type: docs
+---
+
+| Product | Affected Versions | Related Issues | Fixed In |
+| :------------------------- | :------------------ | :---------------- | :------- |
+| {{<product "ysql,ycql">}} | {{<release "2024.2">}} | {{<issue 26666>}}, {{<issue 26860>}} | {{<release "2024.2.3+">}} |
+
+## Description
+
+A memory leak was identified on TServers under two specific conditions:
+
+- For tables with foreign keys if there isn't sufficient write workload on the referenced or parent table, but there is a lot of workload on the referencing or child table.
+
+- For tables with serializable read-only transactions that did not receive any writes.
+
+A typical customer facing symptom due to this issue is likely to be that the system reaches its soft-memory limit, and this'll cause the system to reject requests. If there is a more rapid memory spike, it can also result in an OOM Killer and the YB-TServer process will get restarted.
+
+Consider the following schema, with the parent table and child table.
+
+```sql
+CREATE TABLE parent_table (key int primary key);
+CREATE TABLE child_table (fkey int references parent_table(key));
+
+-- Insert a bunch of rows to parent_table
+
+BEGIN;
+INSERT INTO child_table(fkey) VALUES(..);
+COMMIT;
+```
+
+## Background
+
+The transaction inserts the row to the child table, and the insert generates intents on the child table's tablet. As this insert references the parent tablets row, a lock record is generated corresponding to the row on the parent tablet's row (in order to maintain the foreign key referential integrity). After the transaction commits, these intents can be applied to regular RocksDB and no longer be maintained in the system. In every tablet (parent and child), DocDB apply process adds the transaction that generates intents to a map. This map is used to determine the set of transactions that the transaction loader loads during the tablet bootstrap process and is necessary to ensure that all the transactions necessary for bootstrap are loaded. This map (consisting of transaction IDs), is persisted along with retryable requests structure at the time of transaction log (WAL) rollover, that is used by tablet bootstrap.
```

Comment:
"that is used by tablet bootstrap" - what is used by tablet bootstrap? not clear if this refers to  "retryable requests structure at the time of transaction log (WAL) rollover" or the map. Rephrase?

Reply:
@aishwarya24 , the map is persisted during WAL rollover and used during the tablet bootstrap process.